tests/hash-playwright/tests/browser-plugin.spec.ts

```diff
@@ -0,0 +1,173 @@
+import { sleep } from "@local/hash-isomorphic-utils/sleep";
+// eslint-disable-next-line no-restricted-imports
+import { Page } from "@playwright/test";
+
+import { expect, test } from "./browser-plugin/fixtures";
+import { loginUsingTempForm } from "./shared/login-using-temp-form";
+import { resetDb } from "./shared/reset-db";
+
+test.beforeEach(async () => {
+  await resetDb();
+});
+
+const loggedOutHeaderLocator = "text=Connect to HASH";
+const createAccountButtonLocator = "text=Create a free account";
+const entityTypeSelectorLocator = '[placeholder="Search for types..."]';
+const quickNoteInputLocator = '[placeholder="Start typing here..."]';
+
+export const signOutAndReloadPopup = async ({
+  extensionId,
+  page,
+}: {
+  extensionId: string;
+  page: Page;
+}) => {
+  // Sign out from HASH
+  await page.goto("/");
+  await page.click(`[data-testid="user-avatar"]`);
+  await page.click("text=Sign Out");
+  await sleep(1_000);
+
+  // Confirm the popup has been signed out
+  await page.goto(`chrome-extension://${extensionId}/popup.html`);
+  await expect(page.locator(loggedOutHeaderLocator)).toBeVisible();
+
+  // Sign back in again and confirm the quick note is loaded again
+  await loginUsingTempForm({ page });
+  await page.goto(`chrome-extension://${extensionId}/popup.html`);
+};
```

Comment:
Maybe there should be a check here to see whether the quick note input is present or something like that to confirm the browser extension is in the logged in state? If that's not something we want to do here, maybe the comment needs updating.

Reply:
Yep fair, updated in https://github.com/hashintel/hash/pull/3894/commits/d42c5d017f847664fb1c242c850a327d8e703701